sentry/src/main/java/io/sentry/OutboxSender.java

```diff
@@ -107,89 +108,97 @@ private void processEnvelope(final @NotNull SentryEnvelope envelope, final @Null
+//          if (event == null) {
+//            logger.log(
+//                SentryLevel.ERROR,
+//                "Item %d of type %s returned null by the parser.",
+//                items,
+//                item.getHeader().getType());
+//          } else {
+//            if (envelope.getHeader().getEventId() != null
+//                && !envelope.getHeader().getEventId().equals(event.getEventId())) {
+//              logger.log(
+//                  SentryLevel.ERROR,
+//                  "Item %d of has a different event id (%s) to the envelope header (%s)",
+//                  items,
+//                  envelope.getHeader().getEventId(),
+//                  event.getEventId());
+//              continue;
+//            }
+//            hub.captureEvent(event, hint);
+//            logger.log(SentryLevel.DEBUG, "Item %d is being captured.", items);
+//
+//            if (!waitFlush(hint)) {
+//              logger.log(
+//                  SentryLevel.WARNING,
+//                  "Timed out waiting for event submission: %s",
+//                  event.getEventId());
+//              break;
+//            }
+//          }
+//        } catch (Exception e) {
+//          logger.log(ERROR, "Item failed to process.", e);
+//        }
+//      } else {
+//        // send unknown item types over the wire
+//        final SentryEnvelope newEnvelope =
+//            new SentryEnvelope(
+//                envelope.getHeader().getEventId(), envelope.getHeader().getSdkVersion(), item);
+//        hub.captureEnvelope(newEnvelope, hint);
+//        logger.log(
+//            SentryLevel.DEBUG,
+//            "%s item %d is being captured.",
+//            item.getHeader().getType().getItemType(),
+//            items);
+//
+//        if (!waitFlush(hint)) {
+//          logger.log(
+//              SentryLevel.WARNING,
+//              "Timed out waiting for item type submission: %s",
+//              item.getHeader().getType().getItemType());
+//          break;
+//        }
+//      }
+//
+//      if (hint instanceof SubmissionResult) {
+//        if (!((SubmissionResult) hint).isSuccess()) {
+//          // Failed to send an item of the envelope: Stop attempting to send the rest (an attachment
+//          // without the event that created it isn't useful)
+//          logger.log(
+//              SentryLevel.WARNING,
+//              "Envelope had a failed capture at item %d. No more items will be sent.",
+//              items);
+//          break;
+//        }
+//      }
+//
+//      // reset the Hint to its initial state as we use it multiple times.
+//      if (hint instanceof Resettable) {
+//        ((Resettable) hint).reset();
+//      }
+//    }
+
+    // event and minidump need to be within the same envelope, so right now we are not enriching
+    // the event as a hacky solution, but event is gonna ne symbolicated using the minidump
+    hub.captureEnvelope(envelope, hint);
+    if (!waitFlush(hint)) {
+      return;
     }
```

Comment:
we'd need to special case this if we go ahead with mini dumps, as we need to enrich the event but still sending all the left items within the same envelope